src/ssmDocument/ssm/ssmServer.ts

```diff
@@ -0,0 +1,537 @@
+    }
+})
+
+// Retry schema validation on all open documents
+connection.onRequest(ForceValidateRequest.type, async uri => {
+    return new Promise<JsonLS.Diagnostic[]>(resolve => {
+        const document = documents.get(uri)
+        if (document) {
+            // updateConfiguration()
+            validateTextDocument(document, diagnostics => {
+                resolve(diagnostics)
+            })
+        } else {
+            resolve([])
+        }
+    })
+})
+
+// The content of a text document has changed. This event is emitted
+// when the text document first opened or when its content has changed.
+documents.onDidChangeContent(change => {
+    LimitExceededWarnings.cancel(change.document.uri)
+    triggerValidation(change.document)
+})
+
+// a document has closed: clear all diagnostics
+documents.onDidClose(event => {
+    LimitExceededWarnings.cancel(event.document.uri)
+    cleanPendingValidation(event.document)
+    connection.sendDiagnostics({ uri: event.document.uri, diagnostics: [] })
+})
+
+const pendingValidationRequests: { [uri: string]: NodeJS.Timer } = {}
+const validationDelayMs = 500
+
+function cleanPendingValidation(textDocument: JsonLS.TextDocument): void {
+    const request = pendingValidationRequests[textDocument.uri]
+    if (request) {
+        clearTimeout(request)
+        // tslint:disable-next-line: no-dynamic-delete
+        delete pendingValidationRequests[textDocument.uri]
+    }
+}
+
+function triggerValidation(textDocument: JsonLS.TextDocument): void {
+    cleanPendingValidation(textDocument)
+    pendingValidationRequests[textDocument.uri] = setTimeout(() => {
+        // tslint:disable-next-line: no-dynamic-delete
+        delete pendingValidationRequests[textDocument.uri]
+        validateTextDocument(textDocument)
+    }, validationDelayMs)
+}
+
+function validateTextDocument(
+    textDocument: JsonLS.TextDocument,
+    callback?: (diagnostics: JsonLS.Diagnostic[]) => void
+): void {
+    const jsonDocument = getJSONDocument(textDocument)
+    const documentSettings: JsonLS.DocumentLanguageSettings = { comments: 'error', trailingCommas: 'error' }
+    const respond = (diagnostics: JsonLS.Diagnostic[]) => {
+        connection.sendDiagnostics({ uri: textDocument.uri, diagnostics })
+        if (callback) {
+            callback(diagnostics)
+        }
+    }
+
+    // Validate base on json schema provided for SSM Document
+    languageService.doValidation(textDocument, jsonDocument, documentSettings).then(
+        diagnostics => {
+            setTimeout(() => {
+                const currDocument = documents.get(textDocument.uri)
+                if (currDocument) {
+                    respond(diagnostics) // Send the computed diagnostics to VSCode.
+                }
+            }, 100)
+        },
+        error => {
+            connection.console.error(formatError(`Error while validating ${textDocument.uri}`, error))
+        }
+    )
+}
+
+connection.onDidChangeWatchedFiles(change => {
```

Comment:
Can you specify a type for `changes` to avoid the `no-unsafe-any`?

Reply:
Sure